docs/proposals/perf.md

```diff
@@ -0,0 +1,166 @@
+  * Scalability: potential scaling capacity under different load conditions.
+  * CPU Usage: measure the cpu usage of KubeEdge under different load conditions.
+  * Memory Usage: measure the memory usage of KubeEdge under different load conditions.
+* Performance test can be able to run against both dockerized and un-dockerized version of KubeEdge.
+
+### Non-goals
+
+* To design the specific implementation of single performance test.
+
+## Proposal
+
+### Test Framework
+<img src="../images/perf/perf-test-framework.png">
+
+### Test Tools
+* [Kubemark](https://github.com/kubernetes/kubernetes/tree/master/test/kubemark)
+* [Fortio](https://github.com/fortio/fortio)
+
+### Metrics Tools
+* [Prometheus](https://github.com/prometheus/prometheus)
+* [Grafana](https://github.com/grafana/grafana)
+
+## Test Scenarios
+
+### 1. Edge Nodes join in K8S Cluster
+<img src="../images/perf/perf-edgenodes-join-cluster.png">
+
+Test Cases:
+* Measure Edge Nodes join in K8S Cluster startup time.
+
+  Different numbers of Edge Nodes need be tested.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+  This test case ends with all Edge Nodes are in `Ready` status.
+
+### 2. Create Device/Device Model from Cloud
+<img src="../images/perf/perf-create-device.png">
+
+This scenario is expected to measure the northbound API of KubeEdge.
+
+Test Cases:
+* Measure the latency between K8S API Server and KubeEdge Cloud Part.
+* Measure the throughput between K8S API Server and KubeEdge Cloud Part.
+
+### 3. Report Device Status to Edge
+<img src="../images/perf/perf-report-devicestatus.png">
+
+This scenario is expected to measure the southbound API of KubeEdge.
+
+Test Cases:
+* Measure the latency between KubeEdge Edge Part and device.
+
+  Device numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+* Measure the throughput between KubeEdge Edge Part and device.
+
+  Device numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+As the result of the latency and throughput with different device numbers,
+we can evaluate scalability of devices for KubeEdge Edge Part.
+Measure the capacity of devices can be supported by KubeEdge Edge Part.
+<img src="../images/perf/perf-multi-devices.png">
+
+Different protocols are considered to test between KubeEdge Edge Part and devices.
+E.g. Bluetooth, MQTT, ZigBee, BACnet and Modbus and so on.
+Currenly less than 20ms latency can be accepted in Edge IoT scenario.
+Two kinds of test cases can be adopted: emulators of different devices and actual devices.
+
+### 4. Application Deployment from Cloud to Edge
+<img src="../images/perf/perf-app-deploy.png">
+
+This scenario is expected to measure the performance of KubeEdge from Cloud to Edge.
+
+Test Cases:
+* Measure the pod startup time with docker images in the Edge Node.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20]`.
+
+* Measure the pod startup time without docker images in the Edge Node.
```

Comment:
Probably we should not include image download latency as it's out of this scope.

Reply:
OK. Thanks